README.md

```diff
@@ -21,6 +25,40 @@ To test the complete result from a nix-build:
     $ nix-build
     $ nix-shell --run 'cd result/ && python -m http.server'
 
+### Docker Desktop for Mac
+
+Unfortunately it will not build on macOS natively. 
+But it can be build in a [Docker container on macOS](https://www.docker.com/products/docker-desktop). 
+(You may need to increase the maximum "Disk image size" in Docker's Resources Preferences.)
+
+In your favourite macOS terminal:
+
+    $ docker run -p 8000:8000 -it --name nixos-homepage nixos/nix
```

Comment:
```suggestion
    $ git clone https://github.com/NixOS/nixos-homepage.git
    $ cd nixos-homepage
    $ docker run -v $PWD:/work -w /work -p 8000:8000 -it --name nixos-homepage nixos/nix nix-shell
```

It would be better if git is used from the operating system. I think we can expect this to be installed on macos. I realize that then we have to fix `default.nix` and `shell.nix` a bit (which I already did).

This would avoid running nix-shell in nix-shell, since `fd` and `entr` are already present in development environment of `nixos-homepage`

Reply:
@garbas, thanks. My change puts everything in the container. I agree that it is nicer to have the sources on the macos-side and mount those as a volume in the container as you suggest. I tried this and the `nix-shell` command is successful. So that's great. However, `make` then fails in the container:

```
[...]
nix-build -A packages.x86_64-linux.siteStyles --out-link styles
error: creating symlink from '/work/styles.tmp-16815-189614047' to '/nix/store/d9f5ppqpx1p6cswpg3wlzy5dwy7sdwi5-nixos-homepage-styles': Operation not permitted
make: *** [Makefile:199: styles] Error 1
```
I don't know how to fix this.

Reply:
Actually, I do have ideas on how to fix this, but that involves changing how the build is done and that goes beyond the scope of documenting it in Docker on Mac.

Reply:
Is this maybe because the link already existed because you already tried building it outside docker?

Reply:
No, it is because symlinks between container and volumes mounted on a host are unsupported by Docker.